Synchronization with high number of labels with long strings

cubeddu opened the pull request:
Issues 

- #197
- #180

Switching from GET to POST in fetch calls allows the handling of long labels or large data efficiently. Unlike GET, which has URL length limits, POST carries data in the request body, preventing data truncation and ensuring accurate data transmission. This change resolves issues with long labels in URL queries, eliminating risks tied to URL string parameter limits.

## Description of changes:

- Updated fetch functionality for templated queries to use the POST method.
- Resolved linter and pre-commit issues.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Cole-Greer commented on packages/graph-explorer/src/connector/sparql/SPARQLConnector.ts:
```diff
@@ -24,13 +24,13 @@ import oneHopNeighborsBlankNodesIdsTemplate from "./templates/oneHopNeighbors/on
 
   async fetchSchema(options?: QueryOptions): Promise<SchemaResponse> {
-    const ops = { ...options, disableCache: true };
+    const ops = { ...options, disableCache: true, method: "GET" };
```
Is the fetchSchema request the call to the summary API? If so I suppose there are no concerns with using a GET request for this.

Cole-Greer reviewed (dismissed):
LGTM. As long as there are no concerns with the schema request I think this is good to go.

michaelnchin reviewed (changes requested):
Thanks @cubeddu ! Nice job on finding a workaround for the header limit. Couple of changes before we merge this in...

cubeddu commented:
@michaelnchin PR is ready for review.

michaelnchin reviewed (changes requested):
Thanks @cubeddu ! Requesting a couple more changes per testing of the latest revision rebased on main.

**With IAM Auth enabled DB:**

Gremlin/OC/SPARQL connection sync queries are failing immediately with `AccessDeniedExpection` - requests are likely not being SigV4 signed correctly.

```
{
    "error": {
        "status": 500,
        "message": "\n{\n  \"detailedMessage\": \"Missing Authentication Token\",\n  \"code\": \"AccessDeniedException\",\n  \"requestId\": \"--\"\n}"
    }
}
```

**With non-IAM Auth DB:**

- On both Gremlin and OC connection sync, the initial `summary?mode=detailed` endpoint request fails with `BadRequestException`.

```
{
    "error": {
        "status": 500,
        "message": "\n{\n  \"detailedMessage\": \"Bad route: /pg/statistics/summary\",\n  \"code\": \"BadRequestException\",\n  \"requestId\": \"--\"\n}"
    }
}
```

- Gremlin sync is able to succeed via the fallback schema retrieval queries, but OC schema queries are terminated with different 500 error.

```
{
    "error": {
        "status": 500,
        "message": "\n{\n  \"detailedMessage\": \"Missing 'query' parameter for the request\",\n  \"code\": \"MissingParameterException\",\n  \"requestId\": \"--\"\n}"
    }
}
```

cubeddu commented:
> Thanks @cubeddu ! Requesting a couple more changes per testing of the latest revision rebased on main.
> 
> **With IAM Auth enabled DB:**
> 
> Gremlin/OC/SPARQL connection sync queries are failing immediately with `AccessDeniedExpection` - requests are likely not being SigV4 signed correctly.
> 
> ```
> {
>     "error": {
>         "status": 500,
>         "message": "\n{\n  \"detailedMessage\": \"Missing Authentication Token\",\n  \"code\": \"AccessDeniedException\",\n  \"requestId\": \"--\"\n}"
>     }
> }
> ```
> 
> **With non-IAM Auth DB:**
> 
> * On both Gremlin and OC connection sync, the initial `summary?mode=detailed` endpoint request fails with `BadRequestException`.
> 
> ```
> {
>     "error": {
>         "status": 500,
>         "message": "\n{\n  \"detailedMessage\": \"Bad route: /pg/statistics/summary\",\n  \"code\": \"BadRequestException\",\n  \"requestId\": \"--\"\n}"
>     }
> }
> ```
> 
> * Gremlin sync is able to succeed via the fallback schema retrieval queries, but OC schema queries are terminated with different 500 error.
> 
> ```
> {
>     "error": {
>         "status": 500,
>         "message": "\n{\n  \"detailedMessage\": \"Missing 'query' parameter for the request\",\n  \"code\": \"MissingParameterException\",\n  \"requestId\": \"--\"\n}"
>     }
> }
> ```

@michaelnchin @charlesivie changes are up, I addressed the issue and converted the signing of the headers to a middleware.

michaelnchin reviewed (approved):
LGTM, tested and working well. Thanks @cubeddu !
